src/components/shared/RiskAnalysisInfoSummary.tsx

```diff
@@ -0,0 +1,96 @@
+import { EServiceQueries } from '@/api/eservice'
+import { PurposeQueries } from '@/api/purpose'
+import useCurrentLanguage from '@/hooks/useCurrentLanguage'
+import { List, ListItem, ListItemText, Typography } from '@mui/material'
+import { useQuery } from '@tanstack/react-query'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { SectionContainer } from '../layout/containers'
+
+type RiskAnalysisInfoSummaryProps = {
+  riskAnalysisId: string
+  eServiceId: string
+}
+
+export const RiskAnalysisInfoSummary: React.FC<RiskAnalysisInfoSummaryProps> = ({
+  riskAnalysisId,
+  eServiceId,
+}) => {
+  type QuestionItem = { question: string; answer: string; questionInfoLabel?: string }
+
+  const { t } = useTranslation('purpose', { keyPrefix: 'create.eserviceRiskAnalysisSection' })
+  const currentLanguage = useCurrentLanguage()
+
+  const { data: riskAnalysis } = useQuery({
+    ...EServiceQueries.getEServiceRiskAnalysis(eServiceId!, riskAnalysisId!),
+    enabled: Boolean(eServiceId && riskAnalysisId),
+  })
+  const riskAnalysisTemplate = riskAnalysis?.riskAnalysisForm.answers
+
+  const { data: riskAnalysisConfig } = useQuery({
+    ...PurposeQueries.getRiskAnalysisVersion({
+      riskAnalysisVersion: riskAnalysis?.riskAnalysisForm.version as string,
+      eserviceId: eServiceId!,
+    }),
+    enabled: Boolean(riskAnalysis?.riskAnalysisForm.version && eServiceId),
+  })
+
+  const questions: Array<QuestionItem> = React.useMemo(() => {
+    if (!riskAnalysisTemplate || !riskAnalysisConfig) return []
+
+    // Answers in this form
+    const answerIds = Object.keys(riskAnalysisTemplate)
+
+    // Corresponding questions
+    const questionsWithAnswer = riskAnalysisConfig.questions.filter(({ id }) =>
+      answerIds.includes(id)
+    )
+
+    const answers = questionsWithAnswer.map(({ label, options, id, visualType, infoLabel }) => {
+      const question = label[currentLanguage]
+
+      const questionInfoLabel = infoLabel?.[currentLanguage]
+
+      // Get the value of the answer
+      // The value can be of three types: plain text, multiple options, single option
+      const answerValue = riskAnalysisTemplate[id]
+
+      // Plain text: this value comes from a text field
+      if (visualType === 'text') {
+        return { question, answer: answerValue[0] }
+      }
+
+      // Multiple options: this value comes from a multiple choice checkbox
+      const selectedOptions = options?.filter(({ value }) => answerValue.includes(String(value)))
+      const answer = selectedOptions?.map((o) => o.label[currentLanguage]).join(', ') ?? ''
+
+      return { question, answer, questionInfoLabel }
+    })
+
+    return answers
+  }, [riskAnalysisTemplate, riskAnalysisConfig, currentLanguage])
```

Comment:
This is the fifth time this piece of code is copy pasted. 🥲
Let's make it shared!